### ISSUE TITLE
Can't build a trait from enum supplied as query variable

### ISSUE THREAD

**Issue by coryfklein** [ghostdogpr/caliban#664]
I've defined an enum `Sorting` as a sealed trait:

```
sealed trait Sorting

object Sorting {
  case object Id extends Sorting
  case object Name extends Sorting
}
```

I can use pass this enum in GraphiQL just fine:

```
{
  someResource(sorting: Name) {
    id
    name
  }
}
```

But when I pass this same value by Query Variables then Caliban is unable to parse it, yielding the error `Can't build a trait from input \"Name\"`:

```
query SomeResourceQuery($sorting: Sorting!) {
  someResource(sorting: $sorting) {
    id
    name
  }
}
```

And the query variables:

```
{
  "sorting": "Name"
}
```

Since the query variables must be valid Json, I'm assuming the problem lies in Caliban parsing the string `"Name"` into the actual enum. Or am I just using GraphiQL wrong? The auto-complete inside GraphiQL is suggesting `"Name"` as the correct value in accordance with the graphql schema, so it seems this may be an issue with Caliban instead.

**Comment by coryfklein** [comment]
🤔 I'm on caliban 0.7.0 and it looks like the `dispatch` method related here has been updated since that version. Let me see if an update resolves this.

**Comment by ghostdogpr** [comment]
@coryfklein I just tried it with the latest version and it works fine. It seems it was fixed since 0.7.2 indeed https://github.com/ghostdogpr/caliban/releases/tag/v0.7.2

**Comment by coryfklein** [comment]
🙏 Thank you very much @ghostdogpr. I upgraded to the latest Caliban version yesterday and I'll try this one more time - I expect it to work just fine now.
